editoast/src/models/train_schedule.rs

```diff
@@ -115,6 +115,22 @@ pub struct TrainScheduleDetails {
     pub train_path: i64,
 }
 
+#[derive(Serialize, Debug, PartialEq, Clone, Default)]
+pub struct MechanicalEnergyConsumedBaseEco {
+    pub base: f64,
+    pub eco: Option<f64>,
+}
+
+#[derive(Serialize, Debug, PartialEq, Clone, Queryable)]
+pub struct TrainScheduleWithDetails {
```

Comment:
The structs `TrainScheduleDetails` and `TrainScheduleWithDetails` almost have the same name but different contents/usages. Maybe rename one of them and add a small doc comment to explain how/where they're supposed to be used and what they mean.